client/src/main/java/com/aliyun/emr/rss/client/read/Replica.java

```diff
@@ -77,7 +77,13 @@ public int getNumChunks() {
 
   @Override
   public String toString() {
-    return location.getHost() + ":" + location.getFetchPort();
+    String shufflePartition =
+        String.format("%s:%d %s", location.getHost(), location.getFetchPort(), shuffleKey);
+    if (startMapIndex == 0 && endMapIndex == Integer.MAX_VALUE) {
+      return shufflePartition;
+    } else {
+      return String.format("%s[%d,%d)", shufflePartition, startMapIndex, endMapIndex);
```

Comment:
the map index suffix is added because we need it to distinguish the connection for the same workers in retrying logs.